src/OVAL/oval_component.c

```diff
@@ -1973,6 +2051,159 @@ static oval_syschar_collection_flag_t _oval_component_evaluate_ESCAPE_REGEX(oval
+        free(s);
+        return NULL;
+    }
+    s->str[0] = '\0';
+    s->length = 0;
+    s->capacity = INITIAL_CAPACITY;
+    return s;
+}
+
+static void _string_append(t_string *s, char c)
+{
+    if (s->length + 1 >= s->capacity) {
+        if(realloc(s->str, s->capacity + INITIAL_CAPACITY) == NULL)
+            return;
+        s->capacity += INITIAL_CAPACITY;
+    }
+    s->str[s->length++] = c;
+    s->str[s->length] = '\0';
+}
+
+static char *_glob_to_regex (const char *glob, int noescape)
+{
+    t_string *regex;
+    char * result;
+    char c;
+    int i = 0;
+    int state = START;
+    regex = _string_new();
+    if (regex == NULL) {
+        return NULL;
+    }
+    _string_append(regex, '^');
+    while(1) {
+        c = glob[i++];
+        switch (state) {
+        case START:
+            if (c == '\0') {
+                goto finish;
+            } else if (c == '?') {
+                _string_append(regex, '.');
+            } else if (c == '*') {
+                _string_append(regex, '.');
+                _string_append(regex, '*');
+            } else if (c == '.' || c == '|' || c == '^' || c == '(' || c == ')'
+                    || c == '{' || c == '}' || c == '+' ) {
+                _string_append(regex, '\\');
+                _string_append(regex, c);
+            } else if (c == '[') {
+                state = LEFT_BRACKET;
+            } else if (c == '\\') {
+                if (noescape) {
+                    _string_append(regex, '\\');
+                    _string_append(regex, '\\');
+                } else {
+                    state = ESCAPE;
+                }
+            } else {
+                _string_append(regex, c);
+            }
+            break;
+        case LEFT_BRACKET:
+            _string_append(regex,'[');
+            if (c == '!') {
+                _string_append(regex, '^');
+            } else if (c == '\0'){
+                free(regex->str);
+                free(regex);
+                return NULL;
+            } else {
+                _string_append(regex, c);
+            }
+            state = CLASS;
+            break;
+        case CLASS:
+            if (c == '\\') {
+                if (noescape) {
+                    _string_append(regex, '\\');
+                }
+            } else if (c == ']') {
+                state = START;
+            }
+            _string_append(regex, c);
+            break;
+        case ESCAPE:
+            if (c == '?' || c == '*' || c == '[' || c == ']') {
+                _string_append(regex, '\\');
+                _string_append(regex, c);
+            } else {
+                _string_append(regex, '\\');
+                _string_append(regex, '\\');
+                _string_append(regex, c);
+            }
+            state = START;
+        default:
+            break;
+        }
+    }
+finish:
+    _string_append(regex, '$');
+    result = regex->str;
+    oscap_free(regex);
+    return result;
+}
```

Comment:
The entire section above is indented with spaces.

t_string makes sense but I don't like that it's isolated in this file like this. Are you sure there is nothing in openscap common to use for this? If not, wouldn't it be better to rename t_string to something else and put it in common?